### PR TITLE
[receiver/syslog] Migrate config unmarshaling from yaml to mapstructure

### PR DESCRIPTION
**Pull request by djaglowski**
pkg/stanza historically used yaml for configuration unmarshaling.
The process of migrating to mapstructure was begun long ago but
was never completed. This PR switches syslog's underlying operators
to use mapstructure unmarshaling.

### CODE REVIEW

File: pkg/stanza/operator/input/tcp/tcp.go [hunk 1]
```diff
@@ -59,8 +59,8 @@ func NewConfig(operatorID string) *Config {
 
 // Config is the configuration of a tcp input operator.
 type Config struct {
-	helper.InputConfig `yaml:",inline"`
-	BaseConfig         `yaml:",inline"`
+	helper.InputConfig `mapstructure:",squash" yaml:",inline"`
```
Comment (bogdandrutu) [inline pkg/stanza/operator/input/tcp/tcp.go:62]: Do you want to maintain yaml and json tags as well?

Comment (djaglowski) [reply pkg/stanza/operator/input/tcp/tcp.go:62]: For now, but I would like to remove them as a separate effort. There will be a bunch of related code that can be pulled out too, so I think a separate PR will make sense.

Comment (bogdandrutu) [reply pkg/stanza/operator/input/tcp/tcp.go:62]: sure, just checking the long term commitment that we make here.

File: receiver/syslogreceiver/syslog.go [hunk 2]
```diff
@@ -62,19 +63,35 @@ func (f ReceiverType) BaseConfig(cfg config.Receiver) adapter.BaseConfig {
...
+		cfg.TCP = CreateDefaultTCPConfigBase()
+	} else if componentParser.IsSet("udp") {
+		cfg.UDP = CreateDefaultUDPConfigBase()
 	}
-	return &operator.Config{Builder: inputCfg}, nil
+
+	return componentParser.UnmarshalExact(cfg)
+}
+
+func CreateDefaultTCPConfigBase() *tcp.BaseConfig {
```
Comment (bogdandrutu) [inline receiver/syslogreceiver/syslog.go:91]: do you need this public?

File: receiver/syslogreceiver/syslog.go [hunk 3]
```diff
@@ -62,19 +63,35 @@ func (f ReceiverType) BaseConfig(cfg config.Receiver) adapter.BaseConfig {
...
+	return &tcp.NewConfig("tcp_input").BaseConfig
+}
+
+func CreateDefaultUDPConfigBase() *udp.BaseConfig {
```
Comment (bogdandrutu) [inline receiver/syslogreceiver/syslog.go:95]: do you need this public?

**Review by bogdandrutu — APPROVED** [review]

File: receiver/syslogreceiver/syslog.go (hunk 2 above)
Comment (djaglowski) [reply receiver/syslogreceiver/syslog.go]: I just removed these functions, as they were only used once.

Comment (djaglowski) [reply receiver/syslogreceiver/syslog.go:91]: Correction: These functions are used also for `internal/components/receiver_test.go`, which is why they are exported.

Comment (bogdandrutu) [reply receiver/syslogreceiver/syslog.go:91]: I don't thing they should, we need to limit public API as much as possible.

Comment (bogdandrutu) [reply receiver/syslogreceiver/syslog.go:91]: Especially when only for testing.

Comment (djaglowski) [reply receiver/syslogreceiver/syslog.go:91]: I'll submit another PR shortly

**Pull request merged**